src/hardware/serialport/softmodem.cpp

```diff
@@ -489,7 +489,7 @@ void CSerialModem::DoCommand()
 				foundstr++;
 
 			char* helper = foundstr;
-			helper+=strlen(foundstr);
+			helper += strlen(foundstr) - 1;
```

Comment:
How about replacing lines 491-496 with:
```C++
rtrim(foundstr);
```
?

This way the trimming algorithm would remove all whitespace characters, and not only spaces.